contracts/bridge/router/SynapseRouter.sol

```diff
@@ -0,0 +1,511 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IWETH9.sol";
+import "../interfaces/ISynapseBridge.sol";
+import "../interfaces/ISwapQuoter.sol";
+import "./SynapseAdapter.sol";
+
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @notice SynapseRouter contract that can be used together with SynapseBridge on any chain.
+ * On every supported chain SynapseRouter and SwapQuoter contracts need to be deployed.
+ * Chain pools, that are present in the global BridgeConfig should be added to SwapQuoter.
+ * router.setSwapQuoter(swapQuoter) should be executed to link these contracts.
+ * SynapseRouter should be using the same WETH contract that SynapseBridge is (or will be) using.
+ * All supported bridge tokens should be added to SynapseRouter contract.
+ *
+ * @dev Bridging workflow with SynapseRouter contract.
+ * Suppose `routerO` and `routerD` are SynapseRouter deployments on origin and destination chain respectively.
+ * Suppose user wants to send `tokenIn` on origin chain, and receive `tokenOut` on destination chain.
+ * Suppose for this transaction `bridgeToken` needs to be used.
+ * Bridge token address is `bridgeTokenO` and `bridgeTokenD` on origin and destination chain respectively.
+ * There might or might not be a swap on origin and destination chains.
+ * Following set of actions is required:
+ * 1. originQuery = routerO.getAmountOut(tokenIn, bridgeTokenO, amountIn)
+ * 2. Adjust originQuery.minAmountOut and originQuery.deadline using user defined slippage and deadline
+ * 3. fee = BridgeConfig.calculateSwapFee(bridgeTokenD, destChainId, originQuery.minAmountOut)
```

Comment:
This should change once local bridge config is merged in (posting anyway as reminder)